File: server/src/main/java/by/brawl/ws/newdto/AbstractDto.java
package by.brawl.ws.newdto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.codehaus.jackson.annotate.JsonIgnore;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class AbstractDto implements JsonDto {

    @JsonIgnore
    @Override
    public String asJson() throws JsonProcessingException {
        return new ObjectMapper().writeValueAsString(this);
    }
}

File: server/src/main/java/by/brawl/ws/dto/AbstractDto.kt
package by.brawl.ws.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.codehaus.jackson.annotate.JsonIgnore

@JsonInclude(JsonInclude.Include.NON_NULL)
open class AbstractDto : JsonDto {

    @JsonIgnore
    @Throws(JsonProcessingException::class)
    override fun asJson(): String {
        return ObjectMapper().writeValueAsString(this)
    }
}
